python/ccxt/bybit.py

```diff
@@ -5229,33 +5246,49 @@ def set_margin_mode(self, marginMode, symbol: Optional[str] = None, params={}):
     def change_margin_type(self, symbol, is_cross, leverage, is_long):
         self.load_markets()
         symbol = self.find_symbol(symbol)
-        return self.classify_change_margin(symbol, is_long, is_cross, leverage)
+        enableUnifiedMargin, enableUnifiedAccount = self.is_unified_enabled()
+        if enableUnifiedAccount and not self.is_inverse():
+            # TODO: get from account and avoid _change_margin_type/force_change_margin
+            if is_cross:
+                self.set_leverage(symbol, leverage=leverage)
+                return self._change_margin_type(is_cross)
+            else:
+                return self.classify_change_margin(symbol, is_long, is_cross, leverage, force_change_margin=True)
+        else:
+            return self.classify_change_margin(symbol, is_long, is_cross, leverage)
 
     @staticmethod
     def get_same_direction_position(positions, is_long):
+        no_side_position = None
         for position in positions:
             _is_long = position["is_long"]
-            if is_long == _is_long or _is_long is None or is_long is None:
+            if is_long == _is_long or is_long is None:
                 return position
 
-    def _change_margin_type(self, symbol, is_cross, leverage):
+            # making sure that we go through all position before we chose one without side.
+            if _is_long is None:
+                no_side_position = position
+        return no_side_position
+
+    def _change_margin_type(self, is_cross, symbol=None, leverage=None):
         try:
-            values = self.is_unified_enabled()
-            isUnifiedAccount = self.safe_value(values, 1)
-            if isUnifiedAccount and not self.is_inverse():
-                margin_mode = 'REGULAR_MARGIN' if is_cross else 'ISOLATED_MARGIN'
+            enableUnifiedMargin, enableUnifiedAccount = self.is_unified_enabled()
+            if enableUnifiedAccount and not self.is_inverse():
+                margin_mode = 'cross' if is_cross else 'isolated'
                 return self.set_margin_mode(margin_mode)
-            else:
+            elif symbol and leverage:
                 margin_mode = 'CROSS' if is_cross else 'ISOLATED'
                 params = {'buy_leverage': leverage, 'sell_leverage': leverage}
                 return self.set_derivatives_margin_mode(margin_mode, symbol=symbol, params=params)
+            else:
+                raise NotSupported('Non unified accounts and inverse must use symbol and leverage')
         except NotChanged:
             pass
 
     @staticmethod
     def get_change_margin_input(positions, leverage, same_direction_is_long, is_long):
         long_leverage = short_leverage = leverage
-        if same_direction_is_long is not None and is_long is not None:
```

Comment:
you can elaborate why you removed that?

Reply:
same_direction_is_long can be None in UTA if you have no quantity.

Reply:
I did another small change in there, please take a look

Reply:
just verify same_direction_is_long can't appear None in the normal account, even if the position is empty